### PR TITLE
Plugins Catalog: hide plugins based on the Grafana config

### PR DESCRIPTION
**Pull request by leventebalogh**
Fixes #41074
**Related (backend) PR:** https://github.com/grafana/grafana/pull/41563

**What changed?**
- [x] Updated the api requests to filter out plugins based on the `pluginCatalogHiddenPlugins` config settings
- [x] updated the tests

### CODE REVIEW

File: public/app/features/plugins/admin/api.ts [hunk 1]
```diff
@@ -108,8 +109,9 @@ async function getLocalPluginReadme(id: string): Promise<string> {
 }
 
 export async function getLocalPlugins(): Promise<LocalPlugin[]> {
-  const installed = await getBackendSrv().get(`${API_ROOT}`, { embedded: 0 });
-  return installed;
+  const localPlugins: LocalPlugin[] = await getBackendSrv().get(`${API_ROOT}`, { embedded: 0 });
+
+  return localPlugins.filter(Boolean).filter(isLocalPluginVisible);
```
Comment (jackw) [inline public/app/features/plugins/admin/api.ts:114]: How come we need to do `.filter(Boolean)` here?

**Review by jackw — APPROVED** [review]
LGTM! 🏎️

File: public/app/features/plugins/admin/api.ts (hunk 1 above)
Comment (leventebalogh) [reply public/app/features/plugins/admin/api.ts:114]: Thanks for spotting this mate! 🙏 
Probably something that I didn't manage to clean up during the rebase.

Comment (leventebalogh) [reply public/app/features/plugins/admin/api.ts]: Removed in https://github.com/grafana/grafana/pull/41679/commits/179fdc88bdc69d6e164c4022846e215d0a5b7ad5 👍

**Review by marefr — APPROVED** [review]
LGTM

**Pull request merged**